### PR TITLE
feat: Remove @shopify/polaris-tokens dependency

### LINKED ISSUE #4590: Remove legacy polaris-tokens dependency

### PR DESCRIPTION
**Pull request by aaronccasanova**
Closes #4590

### WHAT is this pull request doing?

This PR removes the `@shopify/polaris-tokens` dependency as tokens are now managed and generated within `@shopify/polaris-react`.

### How to 🎩

🖥 [Local development instructions](https://github.com/Shopify/polaris-react/blob/master/README.md#development)
🗒 [General tophatting guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md)
📄 [Changelog guidelines](https://github.com/Shopify/polaris-react/blob/master/documentation/Versioning%20and%20changelog.md)

<details>
<summary>Copy-paste this code in <code>playground/Playground.tsx</code>:</summary>

```jsx
import React from 'react';
import {Page} from '../src';

export function Playground() {
  return (
    <Page title="Playground">
      {/* Add the code you want to test in here */}
    </Page>
  );
}
```

</details>

### 🎩 checklist

- [ ] Tested on [mobile](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting.md#cross-browser-testing)
- [ ] Tested on [multiple browsers](https://help.shopify.com/en/manual/intro-to-shopify/shopify-admin/supported-browsers)
- [ ] Tested for [accessibility](https://github.com/Shopify/polaris-react/blob/master/documentation/Accessibility%20testing.md)
- [ ] Updated the component's `README.md` with documentation changes
- [ ] [Tophatted documentation](https://github.com/Shopify/polaris-react/blob/master/documentation/Tophatting%20documentation.md) changes in the style guide
- [ ] For visual design changes, ping @ sarahill to update the Polaris UI kit

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -70,7 +70,6 @@
   },
   "dependencies": {
     "@shopify/polaris-icons": "^4.11.0",
-    "@shopify/polaris-tokens": "^3.0.0",
```
Comment (aaronccasanova) [inline package.json:73]: I was expecting to see `yarn.lock` updates in addition to this line being removed.

Note: I am in the process of investigating the following error after running:
`yarn remove @shopify/polaris-react`
Screen Shot 2022-01-05 at 2 36 44 PM

Comment (BPScott) [reply package.json:73]: https://github.com/puppeteer/puppeteer/issues/6622#issuecomment-906476717

Seems updating puppeteer to a newer version should help. The dependency comes in from `@shopify/storybook-a11y-test` which was updated to use puppeteer@^10.4.0  several months ago in https://github.com/Shopify/quilt/pull/2045.

Seems like updating to the latest version of `@shopify/storybook-a11y-test` should help. I expect you'd want to do that on the main branch

**Review by kyledurand — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (aaronccasanova) [reply package.json:73]: Makes sense 👍 I put together a PR in the main branch with these changes: https://github.com/Shopify/polaris-react/pull/4870

Comment (aaronccasanova) [reply package.json:73]: Merged the above PR into `main`, updated `v8` and `v9` branches, and merged `v9` into this PR. Checks are passing now 🙌

**Pull request merged**